crates/control/src/role_assignment.rs

```diff
@@ -372,289 +285,320 @@ impl RoleAssignment {
-                match team_ball {
-                    None => (Role::Loser, true, None),
-                    Some(team_ball) => {
-                        if cycle_start_time
-                            .duration_since(team_ball.last_seen)
-                            .unwrap()
-                            > striker_trusts_team_ball
-                        {
-                            (Role::Loser, true, None)
-                        } else {
-                            (Role::Striker, send_spl_striker_message, Some(team_ball))
-                        }
-                    }
-                }
-            }
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
-        },
-
-        //Striker remains Striker, sends message after timeout
-        (Role::Striker, Some(..), None) => (Role::Striker, send_spl_striker_message, team_ball),
-
-        // Striker got a message (either another Player claims Stiker role or Edge-case of a second Striker)
-        (Role::Striker, Some(..), Some(spl_message)) => match &spl_message.ball_position {
-            None => {
-                // another Striker became Loser, so we claim striker since we see a ball
-                (
-                    Role::Striker,
-                    true,
-                    team_ball_from_seen_ball(detected_own_ball, current_pose, cycle_start_time),
-                )
-            }
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
-        },
-
-        //Loser remains Loser
-        (Role::Loser, None, None) => (Role::Loser, false, team_ball),
+    let striker_trusts_team_ball = |team_ball: BallPosition<Field>| {
+        cycle_start_time
+            .duration_since(team_ball.last_seen)
+            .expect("time ran backwards")
+            <= striker_trusts_team_ball_duration
+    };
 
-        (Role::Loser, None, Some(spl_message)) => match &spl_message.ball_position {
-            None => (Role::Loser, false, None), //edge-case, a striker (which should not exist) lost the ball
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
+    match (current_role, detected_own_ball, event) {
+        // Striker lost Ball
+        (Role::Striker, false, Event::None | Event::Loser) => match team_ball {
```

Comment:
```suggestion
        (Role::Striker, false, Event::Tick | Event::Loser) => match team_ball {
```

just naming that came to mind. what do you think?